Source/WebCore/svg/SVGLengthValue.cpp

```diff
@@ -350,35 +430,59 @@ ExceptionOr<void> SVGLengthValue::setValueAsString(StringView string)
     if (string.isEmpty())
         return { };
 
-    return readCharactersForParsing(string, [&](auto buffer) -> ExceptionOr<void> {
-        auto convertedNumber = parseNumber(buffer, SuffixSkippingPolicy::DontSkip);
-        if (!convertedNumber)
+    // FIXME: SVG2 allows trailing spaces?
+    if (isASCIIWhitespace(string[string.length() - 1]))
+        return Exception { ExceptionCode::SyntaxError };
+
+    auto parserContext = CSSParserContext { SVGAttributeMode };
+    auto parserState = CSS::PropertyParserState {
+        .context = parserContext
+    };
+
+    CSSTokenizer tokenizer(string.toString());
+    auto tokenRange = tokenizer.tokenRange();
+
+    if (auto number = CSSPropertyParserHelpers::MetaConsumer<CSS::Number<>>::consume(tokenRange, parserState, { })) {
+        if (!tokenRange.atEnd())
             return Exception { ExceptionCode::SyntaxError };
 
-        auto lengthType = parseLengthType(buffer);
-        if (lengthType == SVGLengthType::Unknown)
+        m_value = WTFMove(*number);
+        return { };
+    }
+
+    tokenRange = tokenizer.tokenRange();
+    if (auto length = CSSPropertyParserHelpers::MetaConsumer<CSS::LengthPercentage<>>::consume(tokenRange, parserState, { })) {
+        if (!tokenRange.atEnd())
+            return Exception { ExceptionCode::SyntaxError };
+
+        // FIXME: Add support for calc
+        // For now, reject it
```

Comment:
Do we have a test case that’s failing because of this?
```suggestion
        // FIXME: Add support for calculated lengths.
```

Reply:
yes, it's `LayoutTests/svg/css/parse-calc-length.html` — this will be supported in a follow-up PR